src/common/tracer.h

```diff
@@ -144,8 +144,15 @@ struct Tracer {
   jspan add_span(std::string_view span_name, const jspan_context& parent_ctx) { return {}; }
   void init(std::string_view service_name) {}
 };
-  inline void encode(const jspan_context& span, bufferlist& bl, uint64_t f=0) {}
-  inline void decode(jspan_context& span_ctx, ceph::buffer::list::const_iterator& bl) {}
+
+  inline void encode(const jspan_context& span_ctx, bufferlist& bl, uint64_t f = 0) {
+    ENCODE_START(1, 1, bl);
+    // jaeger is missing, set "is_valid" to false.
+    auto is_valid = false;
+    encode(is_valid, bl);
+    ENCODE_FINISH(bl);
+  }
+  inline void decode(jspan_context& span_ctx, bufferlist::const_iterator& bl) {}
```

Comment:
I think `!HAVE_JAEGER` builds should still decode so that compat version is checked just in case and also to avoid admitting an utter garbage bufferlist.